### DISCUSSION TITLE
citre-peek-toggle

### DISCUSSION THREAD

**Discussion by masatake** [universal-ctags/citre#41] [Show and tell]
Do I have to type "M-x citre-peek" every time when I need the peek-window?
I'm using the following code to toggle the window.

```elisp
(defun citre-peek-toggle ()
  (interactive)
  (if citre-peek--ov
      (call-interactively 'citre-peek-abort)
    (call-interactively 'citre-peek)))
(define-key global-map [C-return] 'citre-peek-toggle)
```

**Comment by AmaiKinono** [comment]
> Do I have to type "M-x citre-peek" every time when I need the peek-window?

Yes... I actually don't quite understand this question. You sure need a command to open the window, don't you?

A peek window should always be there unless you press `C-g` or use  `citre-peek-jump`. I'm also planning to make the window follow you after you jump (for cross reading). That's pretty much the life-cycle of a peek window. Just to confirm, does it behave like this on your side?

> I'm using the following code to toggle the window.

You can bind `C-RET` to `citre-peek`, and in `citre-peek-keymap`, bind `C-RET` to `citre-peek-abort` (and maybe unbind `C-g`). This will do the same for you.

`citre-peek` is for peeking the symbol under point. For `toggle`, I'll create a command to restore the recent peek session (I think this is closer to `toggle`).

**Comment by AmaiKinono** [comment]
The new version of `citre-peek` has much more goodies now, so I created a better `toggle` command for you :D

``` elisp
(defun citre-peek-toggle (&optional new)
  "Abort the peek session, or restore if not in a peek session.
When there's no session to restore, or NEW is non-nil (specified
by prefix argument), peek the symbol under point."
  (interactive "P")
  (if citre-peek--mode
      (citre-peek-abort)
    (if (and citre-peek--session-root-list (not new))
        (citre-peek-restore)
      (citre-peek))))

(define-key global-map [C-return] 'citre-peek-toggle)
```

**Reply by masatake** [reply]
Thank you!

**Reply by masatake** [reply]
I tried. This is nice. Really comfortable!

**Reply by AmaiKinono** [reply]
That's the best thing I could hear :)
